routers/web/repo/view.go

```diff
@@ -331,6 +331,16 @@ func renderReadmeFile(ctx *context.Context, subfolder string, readmeFile *git.Tr
 
 		ctx.Data["FileContent"] = buf.String()
 	}
+
+	lfsLock, err := git_model.GetTreePathLock(ctx, ctx.Repo.Repository.ID, path.Join(ctx.Repo.TreePath, readmeFile.Name()))
+	if err != nil {
+		ctx.ServerError("GetTreePathLock", err)
+		return
+	}
```

Comment:
Does this check really makes sense? Even if there is a check .... imagine a case:

1. User A opens the "view page", and see the "Edit" link
2. User A opens the link, and see the editor, editing ....
3. User B lock the LFS file

Then such check doesn't help.

IMO it is the "editor page"'s duty to handle various edge cases, but no the "view page".

Reply:
This logic is just copied from the normal file view page. I don't know if it makes sense, but I think threes a reason that these exists.

https://github.com/go-gitea/gitea/blob/f65977df3a65e193995c0bdf7069553fd5867fe2/routers/web/repo/view.go#L546-L560

Reply:
It looks pretty strange. If `!fInfo.isLFSFile`, then how could `lfsLock != nil` ? Is a non-LFS file able to have a LFS lock?

Reply:
Your right I don't noticed that. So should we just check if it isn't a  file is a LFS file or should we just the if it's locked?

Reply:
IMO "checking if it isn't a file is a LFS file" seems enough.